test/rules/best-practises/no-empty-blocks.js

```diff
@@ -123,4 +183,235 @@ describe('Linter - no-empty-blocks', () => {
     const labelIndex = Math.floor(lastItemIndex / 5) * 4
     return items[labelIndex]
   }
+
+  it('should respect config severity level', () => {
+    const code = 'contract A { }'
+
+    const warnReport = linter.processStr(code, {
+      rules: { 'no-empty-blocks': 'warn' },
+    })
+    assertWarnsCount(warnReport, 1)
+
+    const errorReport = linter.processStr(code, {
+      rules: { 'no-empty-blocks': 'error' },
+    })
+    assertErrorCount(errorReport, 1)
+  })
+
+  describe('Abstract contract cases', () => {
+    it('should not raise warning for abstract contract with virtual functions', () => {
+      const code = `
+        abstract contract Base {
+          function foo() public virtual;
+        }
+      `
+      const report = linter.processStr(code, {
+        rules: { 'no-empty-blocks': 'warn' },
+      })
+      assertNoWarnings(report)
+    })
+  })
+
+  describe('Try-catch variations', () => {
+    const TRY_CATCH_CASES = [
+      funcWith('try foo() { revert(); } catch Error(string memory) { }'),
+      funcWith('try foo() { revert(); } catch (bytes memory) { }'),
+      funcWith('try foo() returns (uint) { } catch Error(string memory) { revert(); }'),
+      funcWith('try foo() {  } catch {  }'),
+    ]
+
+    TRY_CATCH_CASES.forEach((code) => {
+      it(`should detect empty blocks in try-catch: ${label(code)}`, () => {
+        const report = linter.processStr(code, {
+          rules: { 'no-empty-blocks': 'warn' },
+        })
+        const expectedWarnings = code.includes('{  } catch {  }') ? 2 : 1
+        assertWarnsCount(report, expectedWarnings)
+      })
+    })
+  })
+
+  describe('Constructor inheritance', () => {
+    it('should not raise warning for empty constructor with multiple inheritance', () => {
+      const code = `
+        contract C is A, B {
```

Comment:
if having a contract equivalent to this one but written like this:
```
contract C is A(1) B(2) {}
```
is considered valid by the compiler then we should probably add an exception (like this one, not configurable is okay)